File: common/src/main/java/com/reservation/common/response/ApiErrorResponse.java
package com.reservation.common.response;

public record ApiErrorResponse(
	boolean success,
	String code,
	String message,
	Object data // or null, or errors 리스트
) {
	public static ApiErrorResponse of(String code, String message) {
		return new ApiErrorResponse(false, code, message, null);
	}

	public static ApiErrorResponse of(String code, String message, Object data) {
		return new ApiErrorResponse(false, code, message, data);
	}
}

File: core/support/src/main/kotlin/com/reservation/support/response/ApiErrorResponse.kt
package com.reservation.support.response

@JvmRecord
data class ApiErrorResponse(
    val success: Boolean,
    val code: String,
    val message: String,
    val dataOrNull: Any? // or null, or errors 리스트
) {
    companion object {
        @JvmStatic
        fun of(code: String, message: String): ApiErrorResponse {
            return ApiErrorResponse(false, code, message, null)
        }

        fun of(code: String, message: String, dataOrNull: Any?): ApiErrorResponse {
            return ApiErrorResponse(false, code, message, dataOrNull)
        }
    }
}
